src/librustc/mir/repr.rs

```diff
@@ -850,6 +851,16 @@ pub enum BinOp {
     Gt,
 }
 
+impl BinOp {
+    pub fn is_checkable(self) -> bool {
+        use self::BinOp::*;
+        match self {
+            Add | Sub | Mul | Shl | Shr => true,
```

Comment:
Seems to me like a potential source of divergence: you could easily forget to add more operators here when implementing overflow elsewhere in the compiler. Perhaps we should have some central authority on which operators are overflow-checkable?

The central authority would also return false for cases when the overflow checks are disabled by a flag for example, and we could make the odd default branch in LLVM a `bug!` too.